src/spatio_temporal_voxel_layer.cpp

```diff
@@ -720,15 +720,17 @@ void SpatioTemporalVoxelLayer::updateBounds( \
     return;
   }
 
-  boost::recursive_mutex::scoped_lock lock(_voxel_grid_lock);
```

Comment:
If we don't lock here won't it have any impact on voxel grid itself? For e.g. what if someone calls the save grid while voxel grid is getting updated?

Reply:
that's why I did the other option,,, out of prudence
but [ResetGridArea has its own lock](https://github.com/rapyuta-robotics/spatio_temporal_voxel_layer/blob/1cc77c7540b160488dbefe62bf80a750c22425fc/src/spatio_temporal_voxel_grid.cpp#L437), so probably it's safe to remove it. Let's hear @SteveMacenski's opinion

Reply:
I'm not really sure what you're asking. The voxel grid lock is for locking modifications to the openVDB grid resource. The layer mutex is for the layer motifications

Reply:
Sorry, will try to make it clear. I found 2 ways to solve the deadlock explained on the description:
1. [lock costmap mutex](https://github.com/rapyuta-robotics/spatio_temporal_voxel_layer/pull/1/files#diff-1f051dcddd60271eb09793462650f4ac1ab9a84172849431dde1aece6926ee23R725)
2. [remove this lock](https://github.com/rapyuta-robotics/spatio_temporal_voxel_layer/pull/1/files#diff-1f051dcddd60271eb09793462650f4ac1ab9a84172849431dde1aece6926ee23R823)

My question is, which option do you think it's better?

Reply:
oh,,, now I get it; my bad: I don't want to replace voxel locking but add costmap locking on top
fixed now

Reply:
Anyway, we will use current change by now; if all goes fine, I'll open a PR to add it to the upstream repo

Reply:
OK!